src/client.js

```diff
@@ -119,17 +119,18 @@ export class Saturn {
       })
     }
 
-    const fetchPromises = Promise.any(origins.map((origin) => createFetchPromise(origin)))
+    const fetchPromises = Promise.any(nodes.map((node) => createFetchPromise(node)))
 
     let log = {
       startTime: new Date()
     }
 
-    let res, url, controller
+    let res, url, controller, node
     try {
-      ({ res, url, controller } = await fetchPromises)
+      ({ res, url, controller, node } = await fetchPromises)
 
       abortRemainingFetches(controller, controllers)
+      log.nodeId = node.id
       log = Object.assign(log, this._generateLog(res, log), { url })
```

Comment:
the assignment on line `133` will be overwritten by `this._generateLog(res, log)`

perhaps this for line 231?

`log.nodeId = log.nodeId ?? headers.get('saturn-node-id')`

Reply:
agreed, changed